test/transactions/utils/validatePublicKey.js

```diff
@@ -58,24 +58,38 @@ describe('validate public key', () => {
 				validatePublicKey
 					.bind(null, tooShortPublicKey)
 					.should.throw(
-						'Public key 215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452b length differs from the expected 64 hex characters (32 bytes) for a public key.',
+						'Public key 215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452b length differs from the expected 32 bytes for a public key.',
 					);
 			});
 		});
 
-		describe('Given a correct public key', () => {
+		describe('Given an array of public keys with one invalid public key', () => {
+			const publicKeys = [
+				'215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca',
+				'215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca',
+				'215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca',
+				'215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bc',
+			];
+			it('should throw an error', () => {
+				validatePublicKeys
+					.bind(null, publicKeys)
+					.should.throw('Invalid hex string');
+			});
+		});
+
+		describe('Given a valid public key', () => {
 			const publicKey =
 				'215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca';
 			it('should return the public key', () => {
-				validatePublicKey(publicKey).should.be.equal(publicKey);
+				validatePublicKey(publicKey).should.be.true();
 			});
 		});
 
-		describe('Given a correct, all-numeric public key', () => {
+		describe('Given a valid public key', () => {
```

Comment:
Still needs the `all-numeric` part: `Given a valid public key with only numeric characters`?